Reporting/ad-extensions-reporter.js

```diff
@@ -0,0 +1,384 @@
+    }
+
+    // Get report
+    Logger.log("Getting report");
+    if (typeof includeExtensionsWithZeroImpressions === 'undefined') {
+        Logger.log("Warning: options variable includeExtensionsWithZeroImpressions not found. " +
+            "Continuing, and by default including extensions that have had no impressions today."
+        );
+        impressionsCondition = '';
+    } else {
+        if (includeExtensionsWithZeroImpressions === true) {
+            var impressionsCondition = '';
+        } else if (includeExtensionsWithZeroImpressions === false) {
+            var impressionsCondition = 'WHERE Impressions > 0 ';
+        } else {
+            throw ("Value of options variable includeExtensionsWithZeroImpressions must be a boolean, either true or false. " +
+                "Currently, it is: " + includeExtensionsWithZeroImpressions)
+        }
+    }
+
+    var placeholderFeedItemReport = AdsApp.report(
+        'SELECT PlaceholderType, AttributeValues, DisapprovalShortNames ' +
+        'FROM PLACEHOLDER_FEED_ITEM_REPORT ' +
+        impressionsCondition +
+        'DURING TODAY'
+    );
+    var reportRows = placeholderFeedItemReport.rows();
+
+    // Process report data, to prepare for outputting to the spreadsheet
+    Logger.log("Processing report data");
+
+    var extensionsByType = {};
+    while (reportRows.hasNext()) {
+        var row = reportRows.next();
+
+        var placeholderTypeId = row['PlaceholderType']
+        if (!(placeholderTypeId in extensionTypeToPlaceholderTables)) {
+            continue;
+        }
+        if (!(placeholderTypeId in extensionsByType)) {
+            extensionsByType[placeholderTypeId] = {};
+        }
+
+        var jsonAttributes = row['AttributeValues'];
+        var jsonDisapprovalNames = row['DisapprovalShortNames'];
+        if ('' === jsonDisapprovalNames.toString()) {
+            var disapprovalNames = [];
+        } else {
+            var disapprovalNames = JSON.parse(jsonDisapprovalNames);
+        };
+        if (!(jsonAttributes in extensionsByType[placeholderTypeId])) {
+            extensionsByType[placeholderTypeId][jsonAttributes] = {};
+            extensionsByType[placeholderTypeId][jsonAttributes]['Disapprovals'] = disapprovalNames;
+            extensionsByType[placeholderTypeId][jsonAttributes]['FeedItemCount'] = 1;
+        } else {
+            extensionsByType[placeholderTypeId][jsonAttributes]['FeedItemCount'] += 1;
+            for (var i = 0, n = disapprovalNames.length; i < n; i++) {
+                var disapprovalName = disapprovalNames[i];
+                var previouslyFoundDisapprovalNames = extensionsByType[placeholderTypeId][jsonAttributes]['Disapprovals'];
+                if (previouslyFoundDisapprovalNames.indexOf(disapprovalName) === -1) {
+                    extensionsByType[placeholderTypeId][jsonAttributes]['Disapprovals'] =
+                        previouslyFoundDisapprovalNames.concat(disapprovalName);
+                }
+            }
```

Comment:
function?

Reply:
Yes! And here it also simplifies the code naming-wise, because we pass in an object and because it's passed by reference we can call it whatever we like in the function itself.